### ISSUE TITLE
provide documentation how to extend mpm

### ISSUE THREAD

**Issue by riedel** [kdeldycke/meta-package-manager#276]
I have multiple other package managers installed on windows:
* cygwin
* chocolatey
* tlmgr

Those have limited relevance to many users and need to be maintained.

It would be great to find a documentation how to extend mpm. Particularly it would be great if "plugins" could be registered without PRs to the main tree.

**Comment by kdeldycke** [comment]
Plugins is a good idea on paper. But the project is not big enough to justify adding more complexity to support external plugins. A monolithic code base is great to maintain the plethora of platforms and package managers out there. It also opens the possibility of easier refactoring and reducing the cost of maintaining the unittests suite (where the big value is). All in all, it is not unlike the approach taken by the [`yt-dlp`](https://github.com/yt-dlp/yt-dlp) project.

So for now the easiest way to have these new package managers supported is to:
1. fork the project
2. duplicate one package manager definition file in the `/managers` subfolder
3. adapt it to the particularities of the new one
4. fix the code until the unittests passes

That being sais, good news, I just implemented Chocolatey support in v4.5.0 that was released today.
